reco_utils/recommender/surprise/surprise_utils.py

```diff
@@ -0,0 +1,45 @@
+import pandas as pd
+
+
+def compute_predictions(algo, data, usercol='userID', itemcol='itemID'):
+    """
+    Computes predictions of an algorithm from Surprise on the data
+    Args:
+        algo (surprise.prediction_algorithms.algo_base.AlgoBase): an algorithm from Surprise
+        data (pd.DataFrame): the data on which to predict
+        usercol (str): name of the user column
+        itemcol (str): name of the item column
+    Returns:
+        pd.DataFrame: dataframe with usercol, itemcol, prediction
+    """
+    predictions = [algo.predict(row[usercol], row[itemcol]) for (_, row) in data.iterrows()]
+    predictions = pd.DataFrame(predictions)
+    predictions = predictions.rename(index=str, columns={'uid': usercol, 'iid': itemcol, 'est': 'prediction'})
+    return predictions.drop(['details', 'r_ui'], axis='columns')
+
+
+def compute_all_predictions(algo, data, usercol='userID', itemcol='itemID', ratingcol='rating', recommend_seen=False):
+    """
```

Comment:
Can we reuse `compute_predictions`? See my comment in surprise_svd_deep_dive.ipynb

Reply:
similar comment as before, @anargyri  and I should meet to because I'm also refactoring svd for the benchmarks https://github.com/Microsoft/Recommenders/blob/benchmark_feb/notebooks/03_evaluate/comparison_movielens.ipynb

Reply:
See earlier comments why I cannot reuse `compute_predictions` for Surprise.